lib/src/migrators/division.dart

```diff
@@ -0,0 +1,208 @@
+  /// ==, and != as long as the aggressive option is enabled.
+  bool operatesOnNumbers(BinaryOperator operator) =>
+      returnsNumbers(operator) ||
+      operator == BinaryOperator.lessThan ||
+      operator == BinaryOperator.lessThanOrEquals ||
+      operator == BinaryOperator.greaterThan ||
+      operator == BinaryOperator.greaterThanOrEquals ||
+      isAggressive &&
+          (operator == BinaryOperator.equals ||
+              operator == BinaryOperator.notEquals);
+
+  /// Returns true if [node] should be treated as division and migrated.
+  ///
+  /// Warns if division is allowed but it's unclear whether or not all types
+  /// are numeric.
+  bool shouldMigrate(BinaryOperationExpression node) {
+    if (!_isDivisionAllowed && onlySlash(node)) {
+      return false;
+    }
+    if (hasNonNumber(node)) return false;
+    if (_expectsNumericResult || _allNumeric(node)) return true;
+    warn("Could not determine whether this is division", node.span);
+    return false;
+  }
+
+  /// Returns true if [node] is entirely composed of number literals and slash
+  /// operations.
+  bool onlySlash(Expression node) {
+    if (node is NumberExpression) return true;
+    if (node is BinaryOperationExpression) {
+      return node.operator == BinaryOperator.dividedBy &&
+          onlySlash(node.left) &&
+          onlySlash(node.right);
+    }
+    return false;
+  }
+
+  /// Returns true if [node] is believed to always evaluate to a number.
+  bool _allNumeric(Expression node) {
+    if (node is NumberExpression) return true;
+    if (node is ParenthesizedExpression) return _allNumeric(node.expression);
+    if (node is UnaryOperationExpression) return _allNumeric(node.operand);
+    if (node is FunctionExpression) return isAggressive;
+    if (node is BinaryOperationExpression) {
+      return returnsNumbers(node.operator) ||
+          (_allNumeric(node.left) && _allNumeric(node.right));
+    }
+    return false;
+  }
+
+  /// Returns true if [node] contains a subexpression known to not be a number.
+  bool hasNonNumber(Expression node) {
+    if (node is ParenthesizedExpression) return hasNonNumber(node.expression);
+    if (node is BinaryOperationExpression) {
+      return hasNonNumber(node.left) || hasNonNumber(node.right);
+    }
+    return node is BooleanExpression ||
+        node is ColorExpression ||
+        node is ListExpression ||
+        node is MapExpression ||
+        node is NullExpression ||
+        node is StringExpression;
+  }
+
+  /// Adds a patch replacing the operator of [node] with ", ".
+  void patchSlashToComma(BinaryOperationExpression node) {
+    var start = node.left.span.end;
+    var end = node.right.span.start;
+    addPatch(Patch(start.file.span(start.offset, end.offset), ", "));
+  }
+
+  /// Runs [operation] with the given context.
+  void withContext(
+      bool isDivisionAllowed, bool expectsNumericResult, void operation()) {
```

Comment:
```suggestion
      void operation(), {bool isDivisionAllowed, bool expectsNumericResult}) {
```

Right now, if I search for `expectsNumericResult` in this file, the calls to `withContext()` that come up are exactly the ones that *don't* modify `_expectsNumericResult`, since they're the ones where it's being passed as an argument. If you make the parameters named and then do e.g.

```dart
  if (isDivisionAllowed != null) _isDivisionAllowed = isDivisionAllowed;
```

you don't have to explicitly pass the old value, *and* the value that's being modified is named at the call site.